### ISSUE TITLE
Deleting a brew should remove it from Recent Items navbar list

### ISSUE THREAD

**Issue by Gazook89** [naturalcrit/homebrewery#1918]
Issue raised on [reddit](https://old.reddit.com/r/homebrewery/comments/rllw7s/issue_when_i_delete_a_brew_it_still_shows_up_in/).   After having deleted a brew, the brew link still exists in localStorage and pulls up in the Recent Items list from the navbar and navigates to a blank page.

If fixing this, might also make the small CSS change needed to fix #1881

**Comment by ericscheid** [comment]
Some thoughts from gitter:

> I wonder if it's one user across multiple devices/browsers? The delete would/could only update the local storage on the machine that it's run on.
> However pushing Recently Edited/Viewed in the UserInfo object would make these changes account wide.

> Also, if a link is followed that results in 410 Gone or 404 Missing then that link should be removed from the recent brews list too. Maybe.
>
> Or maybe not - my browser (via a plugin) will notice a 404 or 410 and then query the wayback machine for a cached version. Be sad to lose the link to trigger that.

> I've been thinking about this. It seems like the obvious solution (deleted on Laptop1 and then visited on Laptop2, both should now have that entry removed from recent brews). But at the same time I feel link users have hit that error and then relied on that URL to have me recover the brew for them. If we automatically delete it on a visit then they might lose the url?
> 
> Maybe if we improve the "Can't get that" error page and add a button that says "remove from recent brews".

TL;DR: maybe, maybe not.

**Comment by ericscheid** [comment]
Alternative suggestion:

When viewing the Recent Brews list, and hovering over an item, display an icon for removing the item from the list. This would be regardless of whether it is deleted or not. It's just a way to tidy the list.

**Comment by calculuschild** [comment]
I like the idea of an "on-hover" delete button in the list itself. I vote we implement that as the immediate solution to this issue, but also implement the better error page as part of a separate issue: https://github.com/naturalcrit/homebrewery/issues/1924

**Comment by Gazook89** [comment]
> When viewing the Recent Brews list, and hovering over an item, display an icon for removing the item from the list. 

Here is what that looks like on my branch:

| 1 | 2 | 3 | 4 |
|:---:|:---:|:---:|:---:|
|  <img width="129" alt="image" src="https://user-images.githubusercontent.com/58999374/191636612-2c94a0ba-5de9-4a73-8217-7224341757ff.png"> |  <img width="129" alt="image" src="https://user-images.githubusercontent.com/58999374/191635651-682db083-59b4-4dd3-b784-c138179fb8d1.png">  | <img width="129" alt="image" src="https://user-images.githubusercontent.com/58999374/191636306-527daf96-3027-433c-ac9e-ca0c78955657.png"> | *Your entry here* |

- The delete button only shows on hover for each item, but it does obscure the text in the item.
- I hesitate to make the delete button any smaller to 'inline' with the title or below the title.
- The hover state of the delete button itself could be just a lighter gray background
- Right now, an html node for the delete button exists on every recent item, they are just set to `display:none;` until the recent item is hovered and the display turned to `grid` (for convenient `place-content` positioning).  If it is important to only create a Delete Button node when necessary, i'll switch to using an event handler(?)

----
Sidenote:  The :hover state of the recent items adds that blue background (which I like), and it makes the gray timestamp very hard to read.  Should the hover state also change that timestamp color?

**Comment by Gazook89** [comment]
I have a better understanding of how the recentItems list is created now after looking at it more...i'm not sure the previous questions are hugely important.  It's basically going to be number 4.
